### PR TITLE
Reduce logging for etcd caching, and start collecting simple statistics per thread.

### PR DESCRIPTION
**Pull request by mikalstill**

### CODE REVIEW

File: shakenfist/etcd.py [hunk 1]
```diff
@@ -98,7 +113,12 @@ def get(self, key):
             raise exceptions.PrefixNotInCache(
                 'The prefix %s is not in the cache' % key)
 
-        return self.cache.get(key)
+        val = self.cache.get(key)
+        if val:
+            _increment_statistics('cache_miss')
```
Comment (mandoonandy) [inline shakenfist/etcd.py:118]: I think this might be backwards? You have a "hit" when the value is found.

**Comment by mandoonandy** [comment]
This etcd cache will raise an exception if it's queried for a prefix it hasn't cached. So this PR can't count cache misses, but it will be counting actual DB absence. Is that intended?

**Comment by mikalstill** [comment]
Yeah, what I mean by "miss" here is you cached instances, and then asked for an instance which didn't exist. It should be rare I suppose.

**Comment by mikalstill** [comment]
This got a big re-write based on Andy's and my chat about #1106.

**Review by mandoonandy — APPROVED** [review]
LGTM

**Pull request merged**